### PR TITLE
MGMT-8198: Rename host_cluster_registered to host_registration_succeeded

### PR DESCRIPTION
**Pull request by nmagnezi**
# Assisted Pull Request

## Description

This event is only used once in V2RegisterHost with InfraEnv[1].
The rationale for renaming is:
1. Event name should be on par with the respective cluster
   event: cluster_registration_succeeded
2. host_registration_succeeded is on par with host_registration_failed.
3. The event does not include a cluster id.

[1] https://github.com/openshift/assisted-service/blob/c48bcdf24e33c114545089df0ae90fa27b926e24/internal/bminventory/inventory.go#L6273-L6275

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @filanov 
/cc @masayag 
/cc @arielireni  
/cc @avishayt  
## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/events.yaml [hunk 1]
```diff
@@ -615,8 +615,8 @@ x-events:
     host_name: string
     approved_value: bool
 
-- name: host_cluster_registered
-  message: "Host {host_name}: registered to cluster"
+- name: host_registration_succeeded
+  message: "Host {host_name}: Successfully registered"
```
Comment (masayag) [inline docs/events.yaml:619]: no need for cluster-id in this context ?

Comment (nmagnezi) [reply docs/events.yaml:619]: looked at this again.
I saw no `cluster_id` in `V2RegisterHostParams`, however, I do see `V2RegisterHost()` attempts to fetch the cluster via `getBoundCluster()` so I'll add this as an optional param.

Comment (nmagnezi) [reply docs/events.yaml:619]: Done

**Pull request merged**